packages/playwright-core/bin/container_novnc_proxy.js

```diff
@@ -0,0 +1,103 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const http = require('http');
+const { debug, httpProxy, program } = require('../lib/utilsBundle');
+const packageJSON = require('../package.json');
+
+const debugLog = debug('pw:proxy');
+
+const CSS_TYPOGRAPHY = `
+  :root {
+    --monospace: Consolas, Menlo, monospace;
+    --regular: system-ui, -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Droid Sans, Helvetica Neue, Fira Sans, sans-serif;
+    --text-color: #24292e;
+    --background-color: #fff;
+    --link-color: #0366d6;
+    --border-color: #f3e5f5;
+  }
+  body {
+    margin: 0;
+    font: 1em/1.6 var(--regular);
+    font-weight: 400;
+    font-style: normal;
+    text-rendering: optimizeLegibility;
+    -webkit-font-smoothing: antialiased;
+    color: var(--text-color);
+    font-size: 12px;
+  }
+  img, iframe { border: none }
+  h1,h2,h3,h4,h5,h6 {margin: 0.6em 0; font-weight: normal}
+  h1 {font-size: 2.625em; line-height: 1.2}
+  h2 {font-size: 1.625em; line-height: 1.2}
+  h3 {font-size: 1.3125em;line-height: 1.24}
+  h4 {font-size: 1.1875em;line-height: 1.23}
+  h5,h6 {font-size: 1em; font-weight:bold}
+  a { color: var(--link-color); text-decoration: none; }
+  hbox { display: flex; align-items: center; }
+  vbox { display: flex; flex-direction: column; }
+`;
+
+program
+    .command('start')
+    .description('reverse proxy for novnc and playwright server')
+    .option('--port <number>', 'port number')
+    .option('--server-endpoint <url>', 'Playwright Server endpoint')
+    .option('--novnc-endpoint <url>', 'novnc server endpoint')
+    .option('--novnc-ws-path <string>', 'novnc websocket path')
+    .action(async function(options) {
+      launchReverseProxy(options.port, options.serverEndpoint, options.novncEndpoint, options.novncWsPath);
+    });
+
+program.parse(process.argv);
+
+async function launchReverseProxy(port, serverEndpoint, novncEndpoint, novncWSPath) {
+  const vncProxy = httpProxy.createProxyServer({ ws: true, target: novncEndpoint });
+  const serverProxy = httpProxy.createProxyServer({ ws: true, target: serverEndpoint });
+
+  const httpServer = http.createServer((request, response) => {
+    if (request.url === '/' && request.method === 'GET') {
+      response.writeHead(200, {
+        'content-type': 'text/html',
+      }).end(`
+        <style>${CSS_TYPOGRAPHY}</style>
+        <body style='display: flex; align-items: center; justify-content: center; flex-direction: column;'>
+          <h1>Playwright Container</h1>
+          <a href="/screen">View Screen</a>
+        </body>
+      `);
+    } else if ((request.url === '/screen' || request.url === '/screen/') && request.method === 'GET') {
+      response.writeHead(307, {
+        Location: `/screen/?resize=scale&autoconnect=1&path=${novncWSPath}`,
```

Comment:
I want to understand whether this is still secure with the unguessable vnc path, or not :)

Reply:
We played with this offline and seems to be secure!